### PR TITLE
bumping version to 3.10.0

### LINKED ISSUE #4202: Cut new release 3.10
Release early, release often.

We have a couple of nice new features on the docket and our release process is getting more stream-lined.

We do need a release manager, anyone wants to volunteer?

### PR DESCRIPTION
**Pull request by Spaak**
Ready to release, following this checklist: https://github.com/pymc-devs/pymc3/wiki/PyMC3-Release-Checklist

Closes #4202.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release Notes
 
-## PyMC3 3.10.0 (on deck)
+## PyMC3 3.10.0 (7 December 2020)
```
Comment (michaelosthege) [inline RELEASE-NOTES.md:3]: You can add yourself as the release manager here. (As in the releases below.)

For the new "on deck" section I suggest something along the lines of `## PyMC3 latest (on deck)`.

Also, I think we forgot to mention & link some of the latest PRs in the release notes. This list is incomplete:
+ Model logp is now tested before starting to `find_map` or `sample` - surfacing inf/nan problems in the model earlier (see #4211)
+ `model_to_graphviz` now has a `formatting` kwarg that can be used to show more detailed node information (see #4260)

**Review by michaelosthege — CHANGES_REQUESTED** [review]
Just a few things missing from the release notes.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (Spaak) [reply RELEASE-NOTES.md:3]: I'm already listed as release manager (all the way at the bottom--it's a long list of changes ;) ), see #4268.

Checking model logp was mentioned already, but with #4116 (the issue) linked; I agree linking the PR (#4211) is better. I'll also add #4260 to the list of PRs related to the str representations. Other than that I think that's it?

Comment (Spaak) [reply RELEASE-NOTES.md:3]: Oh yes I'll add the new "on deck" after creating the release, as per the [release guidelines](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Release-Checklist).

**Review by twiecki — APPROVED** [review]

**Pull request merged**